.github/workflows/ci-pipeline.yml

```diff
@@ -18,11 +18,27 @@ jobs:
     - run: docker build --target test --tag my-test-image .
     - run: docker run --env-file .env.test  my-test-image
 
+  deploy:
+    name: Build and deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'push' && github.ref == 'refs/heads/exercise-9'
+    steps:
+    - uses: actions/checkout@v2
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - run: docker build --target production --tag kallya21/todo-app:latest .
+    - run: docker push kallya21/todo-app:latest
+    - run: curl -dH -X POST '${{ secrets.AZURE_WEBHOOK_URL }}'
```

Comment:
Nice work getting this in place in the end - for my records I'm assuming this is an unescaped URL?

Reply:
A nice bonus to add to the curl command here can be a `--fail` argument - without this curl reports success if the request gets a response, even if that response is a 4xx/5xx response!

Reply:
Yes, it's an unescaped URL